src/web/components/CMP.tsx

```diff
@@ -5,6 +5,20 @@ import {
 } from '@guardian/consent-management-platform';
 import { ConsentManagementPlatform } from '@guardian/consent-management-platform/lib/ConsentManagementPlatform';
 
+let variant: string | undefined;
+
+const participations = localStorage.getItem('gu.ab.participations');
```

Comment:
Calls to local storage often fail (Safari private mode, quota exceeded, etc.) causing errors to throw so a try catch here would be good. Maybe it could be dropped down into the one below?

I guess you could argue this should be abstracted but I'm not sure we're there yet, abstractions come at a price and I find being able to access things like localStorage directly a big dev ex plus.